includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -1038,6 +1116,66 @@ public function ajax_add_to_cart() {
 		wp_send_json( $data );
 	}
 
+	/**
+	 * Handles payment requests on the Pay for Order page.
+	 *
+	 * @throws Exception All exceptions are handled within the method.
+	 */
+	public function ajax_pay_for_order() {
+		check_ajax_referer( 'pay_for_order' );
+
+		if (
```

Comment:
> I assume that we should return an error response if this statement is false. Am I correct?
My assumption was that if the request does not contain the necessary fields (no verification if anything exists at this point), it probably doesn't originate from the proper place and the response format does not really matter. Anyhow, there is a properly formatted JSON along an error HTTP code.

> Also, WordPress request validation can be used to perform validation instead of using it like this. It will be more clear IMHO.
What do you mean by WordPress request validation? [Core docs](https://developer.wordpress.org/plugins/security/data-validation/#built-in-php-functions) describe that for basic stuff like this, we should use PHP functions.

> Moreover, $_POST['payment_method'] and $_POST['wcpay-payment-method']is sent via POST, but it is never used in this function. Is there a reason behind it?

`payment_method` is used here to verify that this is a WCPay request. `wcpay-payment-method` is one of the few request parameters here, and is later used by the gateway's `process_payment` method.